Userland/Services/WindowServer/Overlays.cpp

```diff
@@ -230,16 +231,77 @@ WindowGeometryOverlay::WindowGeometryOverlay(Window& window)
     update_rect();
 }
 
+void WindowGeometryOverlay::set_actual_rect()
+{
+    if (m_move_into_overlay_rect_animation.animation) {
+        auto const& starting_rect = m_move_into_overlay_rect_animation.starting_rect.value();
+        m_move_into_overlay_rect_animation.current_rect = starting_rect.interpolated_to(starting_rect.centered_within(m_move_into_overlay_rect_animation.tile_overlay_rect_at_start), m_move_into_overlay_rect_animation.progress);
+        set_rect(m_move_into_overlay_rect_animation.current_rect);
+    } else {
+        set_rect(m_ideal_overlay_rect);
+    }
+}
+
+void WindowGeometryOverlay::start_or_stop_move_to_tile_overlay_animation(TileWindowOverlay* tile_window_overlay)
+{
+    if (!tile_window_overlay) {
+        if (m_move_into_overlay_rect_animation.animation)
+            m_move_into_overlay_rect_animation.animation->stop();
+        m_move_into_overlay_rect_animation = {};
+        return;
+    }
+
+    auto tile_overlay_rect = tile_window_overlay->tiled_frame_rect();
+    if (m_move_into_overlay_rect_animation.tile_overlay_rect_at_start != tile_overlay_rect || !m_move_into_overlay_rect_animation.starting_rect.has_value()) {
+        if (!m_move_into_overlay_rect_animation.starting_rect.has_value()) {
+            // The tile overlay was just started on one location, we want to move the geometry overlay from the center of the window to the center of the tile overlay
+            m_move_into_overlay_rect_animation.starting_rect = m_ideal_overlay_rect;
+            m_move_into_overlay_rect_animation.current_rect = m_ideal_overlay_rect;
+        } else if (m_ideal_overlay_rect.size() != m_move_into_overlay_rect_animation.starting_rect.value().size()) {
+            // The geometry label size was changed. This normally would only happen when the text changes while moving the window around. But because
```

Comment:
very nit: Consider putting linebreaks in places that make semantic sense. Having `But because` hanging at the end of this line instead of the start of the next looks weirder than if it was at the start of the next imho. (I realize this is far out in bikeshed territory – feel free to ignore ofc)